protocol/nats_jetstream/v2/protocol.go

```diff
@@ -67,6 +86,37 @@ func NewProtocolFromConn(conn *nats.Conn, stream, sendSubject, receiveSubject st
 	return p, nil
 }
 
+func NewProtocolFromConnV2(conn *nats.Conn, stream, sendSubject string, jsOpts []jetstream.JetStreamOpt, opts ...ProtocolOption) (*Protocol, error) {
```

Comment:
@evankanderson what do you think about the V2 change? We have to trade backwards-compatibility with maintenance and simple APIs.

@stephen-totty-hpe 
If we go down this route, we need to mark the other method(s) deprecated so linters can catch it. Also, should we use this as opportunity to also add `context.Context` as first parameter to the API?

Reply:
BTW, I have other PRs to modify behavior in the existing implementation that are not needed using the implementation I have here with the jetstream package.  So however it is done, the new jetstream library would be a welcome change.  For instance, the new jetstream consumers are pull consumers.  Also, an array of filter subjects are exposed.  An there are error handlers that can be passed through the "config" objects.